src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,29 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executor.submit(() -> listener.onEvent(event));
+        }
     }
 
     public void shutdown() {
-
+        executor.shutdown();
     }
```

Comment:
When shutting down the ExecutorService, it's good practice to first call shutdown() to stop accepting new tasks, and then call awaitTermination() with a timeout to ensure all tasks have completed. This avoids potential resource leaks.